src/compiler/checker.ts

```diff
@@ -14400,12 +14432,126 @@ namespace ts {
             }
         }
 
+        function checkUnusedLocals(node: FunctionDeclaration | MethodDeclaration | ConstructorDeclaration | FunctionExpression | ArrowFunction | ForInStatement | Block): void {
+            checkUnusedIdentifiers(node);
+            checkUnusedParameters(node);
+        }
+
+        function checkUnusedIdentifiers(node: FunctionDeclaration | MethodDeclaration | ConstructorDeclaration | FunctionExpression | ArrowFunction | ForInStatement | Block): void {
+            if (compilerOptions.noUnusedLocals && !isInAmbientContext(node)) {
+                for (const key in node.locals) {
+                    if (hasProperty(node.locals, key)) {
+                        const local = node.locals[key];
+                        if (!local.hasReference && local.valueDeclaration && local.valueDeclaration.kind) {
+                            if (local.valueDeclaration.kind !== SyntaxKind.Parameter) {

```

Comment:
i would just inline this in previous condition.`!local.hasReference && local.valueDeclaration && local.valueDeclaration.kind !== SyntaxKind.Parameter`